core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -562,6 +562,14 @@ public ActualProperties visitExchange(ExchangeNode node, List<ActualProperties>
             // are computed for local exchanges.
             // TODO: implement full properties for local exchanges
             if (node.getScope() == LOCAL) {
```

Comment:
Is local gathering exchange `isEffectivelySingleStream`? I'm thinking of a plan
```
1. Limit[with pre-sorted input]
  |
2. Exchange[RoundRobin]
  |
3. GatheringMergeSortExchange
```

I think 2) might cut ordering properties as local gathering exchange might not be `isEffectivelySingleStream`

Reply:
Local gathering exchange does not seem to be `isEffectivelySingleStream`. It results in below properties
```
ActualProperties{globalProperties=Global{nodePartitioning=Optional.empty, streamPartitioning=Optional.empty, nullsAndAnyReplicated=false}, localProperties=[S↑→(col_b)], constants={}}
```
isSingleStream is true here though.
Is such a plan likely in practise ? I think the limit (before TopN is created) would get pushed down and merged with the lower sort in such scenario.

Reply:
Is it possible that local merge sort gather produces multiple streams? Or should we just update `PropertyDerivations` to report it as `effectivelySingleStream`?

Reply:
> Is it possible that local merge sort gather produces multiple streams?

local merge sort produces single input. 

> Or should we just update PropertyDerivations to report it as effectivelySingleStream?

I think we should do that

Reply:
I made a change now to make local merging exchange effectivelySingleStream, PTAL